contracts/AnchoredPriceView.sol

```diff
@@ -0,0 +1,302 @@
+            priceSixDecimals = oneDollar;
+        } else if (cToken == cWbtcAddress || cToken == cEthAddress) {
+            priceSixDecimals = prices[getOracleKey(cToken)];
+        } else {
+            uint256 usdPerEth = prices["ETH"];
+            uint256 ethPerToken = anchor.getUnderlyingPrice(cToken);
+            // usdPerEth - 6 decimals
+            // ethPerStandardErc20 - 18 decimals
+            // divide by 1e18 to drop down to 6 decimals
+            priceSixDecimals = mul(usdPerEth, ethPerToken) / 1e18;
+        }
+
+        // comptroller expects price to have 18 decimals,
+        // and additionally upscaled by 1e18 - underlyingdecimals
+        // base decimals is 1e6, so start by addint twelve
+        uint256 additionalScale;
+        if ( cToken == cUsdcAddress || cToken == cUsdtAddress )  {
+            additionalScale = 1e24;
+        } else if ( cToken == cWbtcAddress )  {
+            additionalScale = 1e22;
+        } else {
+            additionalScale = 1e12;
+        }
+
+        return mul(priceSixDecimals, additionalScale);
+    }
+
+    /**
+     * @notice Returns the cToken address for symbol
+     * @param symbol The symbol to map to cToken address
+     * @return The cToken address for the given symbol
+     */
+    function getCTokenAddress(string memory symbol) public view returns (address) {
+        bytes32 symbolHash = keccak256(abi.encodePacked(symbol));
+        if (symbolHash == symbolEth) return cEthAddress;
+        if (symbolHash == symbolUsdc) return cUsdcAddress;
+        if (symbolHash == symbolDai) return cDaiAddress;
+        if (symbolHash == symbolRep) return cRepAddress;
+        if (symbolHash == symbolWbtc) return cWbtcAddress;
+        if (symbolHash == symbolBat) return cBatAddress;
+        if (symbolHash == symbolZrx) return cZrxAddress;
+        if (symbolHash == symbolSai) return cSaiAddress;
+        if (symbolHash == symbolUsdt) return cUsdtAddress;
+        revert("Unknown token symbol");
+    }
+
+    /**
+     * @notice Returns the symbol for cToken address
+     * @param cToken The cToken address to map to symbol
+     * @return The symbol for the given cToken address
+     */
+    function getOracleKey(address cToken) public view returns (string memory) {
+        if (cToken == cEthAddress) return "ETH";
+        if (cToken == cUsdcAddress) return "USDC";
+        if (cToken == cDaiAddress) return "DAI";
+        if (cToken == cRepAddress) return "REP";
+        if (cToken == cWbtcAddress) return "BTC";
+        if (cToken == cBatAddress) return "BAT";
+        if (cToken == cZrxAddress) return "ZRX";
+        if (cToken == cSaiAddress) return "SAI";
+        if (cToken == cUsdtAddress) return "USDT";
+        revert("Unknown token address");
+    }
+
+    function mul(uint256 a, uint256 b) internal pure returns (uint256) {
+        if (a == 0) {
+            return 0;
+        }
+
+        uint256 c = a * b;
+        require(c / a == b, "SafeMath: multiplication overflow");
```

Comment:
Probably we can drop the "SafeMath:" and possibly pass in message like we sometimes do? Generic message is also fine if/since we have traces now